apm-agent-plugins/apm-quartz-job-plugin/src/test/java/co/elastic/apm/agent/quartz/job/JobTransactionNameInstrumentationTest.java

```diff
@@ -0,0 +1,222 @@
+        tracer = new ElasticApmTracerBuilder()
+                .configurationRegistry(SpyConfiguration.createSpyConfig())
+                .reporter(reporter)
+                .build();
+        ElasticApmAgent.initInstrumentation(tracer, ByteBuddyAgent.install(),
+                Collections.singletonList(new JobTransactionNameInstrumentation(tracer)));
+    }
+
+    @Test
+    void testJobWithGroup() throws SchedulerException, InterruptedException {
+        reporter.reset();
+        resetCounter();
+        Scheduler scheduler=new StdSchedulerFactory().getScheduler();
+        scheduler.clear();
+        JobDetail job = JobBuilder.newJob(TestJob.class)
+    			.withIdentity("dummyJobName", "group1").build();
+        Trigger trigger = TriggerBuilder
+        		.newTrigger()
+        		.withIdentity("myTrigger")
+        		.withSchedule(
+        				SimpleScheduleBuilder.repeatSecondlyForTotalCount(2, 1))
+        		.build();
+        scheduler.scheduleJob(job, trigger);
+        scheduler.start();
+        Thread.sleep(2500);
+        scheduler.shutdown();
+        assertThat(reporter.getTransactions().size()).isEqualTo(getInvocationCount());
+        assertThat(reporter.getTransactions().get(0).getContext().getCustom("scheduledTime")).asString().isNotEmpty();
+        assertThat(reporter.getTransactions().get(0).getContext().getCustom("trigger")).asString()
+        	.isEqualToIgnoringCase(String.format("%s.%s", trigger.getKey().getGroup(), trigger.getKey().getName()));
+        assertThat(reporter.getTransactions().get(0).getName())
+        	.isEqualToIgnoringCase(String.format("%s.%s", job.getKey().getGroup(), job.getKey().getName()));
+    }
+
+    @Test
+    void testJobWithoutGroup() throws SchedulerException, InterruptedException {
+        reporter.reset();
+        resetCounter();
+        Scheduler scheduler=new StdSchedulerFactory().getScheduler();
+        scheduler.clear();
+        JobDetail job = JobBuilder.newJob(TestJob.class)
+    			.withIdentity("dummyJobName").build();
+        Trigger trigger = TriggerBuilder
+        		.newTrigger()
+        		.withIdentity("myTrigger")
+        		.withSchedule(
+        				SimpleScheduleBuilder.repeatSecondlyForTotalCount(2, 1))
+        		.build();
+        scheduler.scheduleJob(job, trigger);
+        scheduler.start();
+        Thread.sleep(2500);
+        scheduler.shutdown();
+        assertThat(reporter.getTransactions().size()).isEqualTo(getInvocationCount());
+        assertThat(reporter.getTransactions().get(0).getContext().getCustom("scheduledTime")).asString().isNotEmpty();
+        assertThat(reporter.getTransactions().get(0).getContext().getCustom("trigger")).asString()
+        	.isEqualToIgnoringCase(String.format("%s.%s", trigger.getKey().getGroup(), trigger.getKey().getName()));
+        assertThat(reporter.getTransactions().get(0).getName())
+        	.isEqualToIgnoringCase(String.format("%s.%s", job.getKey().getGroup(), job.getKey().getName()));
```

Comment:
Add a method for all these common assertions, where transaction type is asserted as well